scipy/signal/tests/test_bsplines.py

```diff
@@ -144,34 +144,82 @@ def test_qspline1d_eval(self):
         assert_allclose(bsp.qspline1d_eval(cj, newx, dx=dx, x0=x[0]), newy)
 
 
-def test_sepfir2d_invalid_filter():
-    filt = np.array([1.0, 2.0, 4.0, 2.0, 1.0])
-    image = np.random.rand(7, 9)
-    # No error for odd lengths
-    signal.sepfir2d(image, filt, filt[2:])
-
-    # Row or column filter must be odd
-    with pytest.raises(ValueError, match="odd length"):
-        signal.sepfir2d(image, filt, filt[1:])
-    with pytest.raises(ValueError, match="odd length"):
-        signal.sepfir2d(image, filt[1:], filt)
-
-    # Filters must be 1-dimensional
-    with pytest.raises(ValueError, match="object too deep"):
-        signal.sepfir2d(image, filt.reshape(1, -1), filt)
-    with pytest.raises(ValueError, match="object too deep"):
-        signal.sepfir2d(image, filt, filt.reshape(1, -1))
-
-def test_sepfir2d_invalid_image():
-    filt = np.array([1.0, 2.0, 4.0, 2.0, 1.0])
-    image = np.random.rand(8, 8)
-
-    # Image must be 2 dimensional
-    with pytest.raises(ValueError, match="object too deep"):
-        signal.sepfir2d(image.reshape(4, 4, 4), filt, filt)
-
-    with pytest.raises(ValueError, match="object of too small depth"):
-        signal.sepfir2d(image[0], filt, filt)
+class TestSepfir2d:
+    def test_sepfir2d_invalid_filter(self):
```

Comment:
Review note: if one adds `?w=1` to the URL of the diff UI, this nicely shows that this test stays unchanged.